Form-Controls/index.html

```diff
@@ -3,26 +3,49 @@
 <head>
   <meta charset="utf-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
-  <title>My form exercise</title>
-  <meta name="description" content="">
+  <title>Form Exercise</title>
+  <meta name="description" content="This form is for collecting user data">
+  <title>Form Exercise</title>
+  <meta name="description" content="This form is for collecting user data">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="stylesheet" href="styles.css">
 </head>
 <body>
-  <header>
-    <h1>Product Pick</h1>
-  </header>
-  <main>
-  <form>
-    <!-- write your html here-->
-    <!-- try writing out the requirements first-->
+
+  <form id="tshirt-order-form" aria-labelledby="form-title">
+    <h1 id="form-title">Order Your T-Shirt</h1>
+
+    <label for="name">Name:</label>
+    <input type="text" id="name" name="name" required minlength="2" placeholder="Your Name">
+
+    <label for="email">Email:</label>
+    <input type="email" id="email" name="email" required placeholder="example@domain.com">
+
+    <fieldset>
+      <legend>Select T-Shirt Color</legend>
+      <label><input type="radio" name="color" value="Red" required> Red</label>
+      <label><input type="radio" name="color" value="Green"> Green</label>
+      <label><input type="radio" name="color" value="Blue"> Blue</label>
+    </fieldset>
+
+    <label for="size">Size:</label>
+    <select id="size" name="size" required>
+      <option value="">Select Size</option>
+      <option value="XS">XS</option>
+      <option value="S">S</option>
+      <option value="M">M</option>
+      <option value="L">L</option>
+      <option value="XL">XL</option>
+      <option value="XXL">XXL</option>
+    </select>
+
+   
+    <label for="delivery-date">Preferred Delivery Date (within 5 working days):</label>
+    <input type="date" id="delivery-date" name="delivery-date" required>
```

Comment:
Date should be from a constrained date range. so don't remove min and max, rather add dates in them.